delocate/libsana.py

```diff
@@ -51,16 +58,71 @@ def tree_libs(start_path, filt_func = None):
             depending_libpath = realpath(pjoin(dirpath, base))
             if not filt_func is None and not filt_func(depending_libpath):
                 continue
+            rpaths = get_rpaths(depending_libpath)
             for install_name in get_install_names(depending_libpath):
                 lib_path = (install_name if install_name.startswith('@')
                             else realpath(install_name))
+                lib_path = resolve_rpath(lib_path, rpaths,
+                                         copy_filt_func)
                 if lib_path in lib_dict:
                     lib_dict[lib_path][depending_libpath] = install_name
                 else:
                     lib_dict[lib_path] = {depending_libpath: install_name}
     return lib_dict
 
 
+def resolve_rpath(lib_path, rpaths, copy_filt_func):
+    """ Return `lib_path` with its `@rpath` resolved
+
+    If the `lib_path` does't have `@rpath` then it's returned as is.
+
+    If `lib_path` has `@rpath` then returns the first `rpaths`/`lib_path`
```

Comment:
I'm not sure this logic is right.  Imagine a library that depends on `@rpath/foo.dylib`.  The rpath contains `/path1, /path2` both of which contain `foo.dylib`. `copy_filt_func` skips `/path1/foo.dylib` for some reason.  Now the returned lib path is `/path2/foo.dylib` even though this is not in fact the `foo.dylib` that the library will link to at run time.

I think `copy_filt_func` is saying, "don't copy stuff from here", but not "don't link to stuff from here".  If you see what I mean.